update to docs  for extract1d for IFU data

jemorrison opened the pull request:
Documentation updates made in JP-1736.

drlaw1558 commented on docs/jwst/references_general/apcorr_reffile.inc:
```diff
@@ -263,18 +240,55 @@ The row selection criteria for each instrument+mode are:
++============+=======+=============+===========+============+==========+
+| MIRI       | MRS   | wavelength  | float     | 1D array   | micron   |
++            +       +-------------+-----------+------------+----------+
+|            |       | radius      | float     | 1D array   | arcsec   |
```
radius, apcor, and apcorr_err are all 2d arrays.  At the moment the 2nd dimension doesn't add information, but in future it could be used to provide different aperture corrections for different radii.

drlaw1558 commented on docs/jwst/extract_1d/description.rst:
```diff
@@ -239,25 +239,27 @@ direction perpendicular to dispersion.
+limits of an annular background aperture.
+The extraction size parameters are given in units of arcseconds and converted to units of pixels
+in the extraction process. 
+
```
Is it worth mentioning the axis_ratio and axis_pa vectors?  These are in the file, but are just placeholders for potential future functionality.

drlaw1558 commented:
Might also be worth clarifying that for IFU mode the extraction location is either the RA/DEC target location indicated by the header (point sources), the center of the cube if the header target location is undefined (point sources) or the entire FOV (extended sources).  I.e., the --use_source_posn doesn't work (see https://jira.stsci.edu/browse/JP-1750)

jemorrison commented:
@hbushouse 
I would like to refer to the table in srctype step on single source observations when explaining how use_source_posn is set. How can I do that instead of just putting in the link to src type step :ref:`srctype <srctype_step> ?
Or maybe you have a better idea.

hbushouse commented:
> @hbushouse
> I would like to refer to the table in srctype step on single source observations when explaining how use_source_posn is set. How can I do that instead of just putting in the link to src type step :ref:`srctype <srctype_step> ?
> Or maybe you have a better idea.

If you add a tag/label to /docs/jwst/srctype/description.rst just before the table, like `.. _srctype_table:` (or whatever you want to call it), you can then reference it in the extract_1d doc using `:ref:`srctype_table``.

jemorrison commented:
@skendrew 
I updated the documentation for extract 1d for some IFU work. I added some clarification here for the use_source_posn and added a link to the srctype table. Is this good or do you want more description  ?

skendrew commented:
@jemorrison the documentation update looks good to me!

hbushouse commented on docs/jwst/extract_1d/description.rst:
```diff
@@ -239,25 +247,27 @@ direction perpendicular to dispersion.
-For IFU cube data, 1D extraction is contolled by a different set of EXTRACT1D
-reference file parameters.
-Note that for an extended source, anything specified in the reference file
+For IFU cube data, 1D extraction is controlled by a different set of EXTRACT1D
```
This phrase seems to end unexpectedly and doesn't flow to the wording in the next line

hbushouse commented on docs/jwst/extract_1d/description.rst:
```diff
@@ -239,25 +247,27 @@ direction perpendicular to dispersion.
+target location is undefined. For extended source data, anything specified in the reference file
 or step arguments will be ignored; the entire image will be
-extracted, and no background subtraction will be done.
+extracted, and no background subtraction will be done. For IFU data the ``use_source_posn`` option is not used. 
```
Really? Why isn't use_source_posn used when we know the data are using a point-source nod pattern? If the hardwired position of the extraction region in the ref file is set to the center of the cube, isn't that going to lead to bad extractions when nodding/dithering is involved and hence the source is not centered spatially in the cube?

jemorrison replied:
The center of the extraction region is set by the target ra/dec and if that is not given then it is set by the center of the IFU. If this method does not work for nodded IFU data then we will need to update the ifu.py routine in extract.py. Nothing about use_source_posn is used in the routine.
@drlaw1558 comments ? I think the extract ifu.py is optimized for MIRI (does MIRI MRS even use Nodding ?)

hbushouse replied:
OK, what you described is basically the same functionality as use_source_posn. For non-IFU modes, when use_source_posn=True, the center of the extraction is computed from the target RA/Dec, and when use_source_posn=False the position is based on the params from the extract1d ref file. So I guess we're OK. It just might be confusing to users that use_source_posn only applies to non-IFU data.

jemorrison replied:
I was confused myself how the extraction parameters for different modes was working. It took me a while to understand it (and I am still learning). One reason why I explicitly said 'use_source_posn was not used in IFU data' was so that the user would not use it. Or try to use it and wonder why it did nothing.  Maybe we need to revisit this with NIRSpec Nodded IFU data. I will leave that to the NIRSpec experts.

hbushouse replied:
If, as you said above, the center of the extraction region is set by the target ra/dec, that should work equally well for both MRS and NRS IFU dithers/nods, because it will force the extraction aperture to follow the source around the field of view. Unless of course what you described above is only being done for MRS data and not for NRS IFU, in which case that would be a problem. I would hope/assume that both types of IFU data are treated the same way.

hbushouse commented on docs/jwst/references_general/apcorr_selection.inc:
```diff
@@ -4,12 +4,12 @@ CRDS selects appropriate APCORR references based on the following keywords.
 NIRISS     INSTRUME, EXP_TYPE, DATE-OBS, TIME-OBS
-NIRSpec    INSTRUME, EXP_TYPE, DATE-OBS, TIME-OBS
-========== ======================================
+NIRSpec    INSTRUME, EXP_TYPE, FILTER, GRATING,DATE-OBS, TIME-OBS
```
need a whitespace following the comma in "GRATING,DATE-OBS"

jemorrison commented:
both IFU modes are treated the same way. Should I remove the line "For IFU data the 'use_source_posn` option is not used". Is that making it confusing ?

________________________________
From: Howard Bushouse <notifications@github.com>
Sent: Wednesday, February 17, 2021 5:43 AM
To: spacetelescope/jwst <jwst@noreply.github.com>
Cc: Jane Morrison <morrison@as.arizona.edu>; Mention <mention@noreply.github.com>
Subject: [EXT]Re: [spacetelescope/jwst] update to docs for extract1d for IFU data (#5642)


External Email

External Email

@hbushouse commented on this pull request.

________________________________

In docs/jwst/extract_1d/description.rst<https://github.com/spacetelescope/jwst/pull/5642#discussion_r577580104>:

>  or step arguments will be ignored; the entire image will be
-extracted, and no background subtraction will be done.
+extracted, and no background subtraction will be done. For IFU data the ``use_source_posn`` option is not used.


If, as you said above, the center of the extraction region is set by the target ra/dec, that should work equally well for both MRS and NRS IFU dithers/nods, because it will force the extraction aperture to follow the source around the field of view. Unless of course what you described above is only being done for MRS data and not for NRS IFU, in which case that would be a problem. I would hope/assume that both types of IFU data are treated the same way.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/spacetelescope/jwst/pull/5642#discussion_r577580104>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AENCOWJEBFX5E6MMV5RING3S7O2XJANCNFSM4WJM6DQQ>.

hbushouse commented:
No, I would not remove that line, because it's important for the users to know that that param has no effect on processing IFU data, so perhaps just rearrange the paragraph such that it starts with that line and then goes on to explain what happens instead.  e.g. something like "The `use_source_posn` parameters is not used when extracting spectra from an IFU cube. Instead, the extraction aperture is centered at the RA/Dec of the target, ...."
